src/plugins/intel_cpu/tests/functional/subgraph_tests/src/remove_reorder.cpp

```diff
@@ -0,0 +1,133 @@
+    std::vector<ov::Shape>
+>;
+
+class RemoveUselessReorderCPUTest: public testing::WithParamInterface<RemoveUselessReorderCPUTestParams>,
+                                 virtual public SubgraphBaseTest {
+public:
+    static std::string getTestCaseName(testing::TestParamInfo<RemoveUselessReorderCPUTestParams> obj) {
+        ElementType inType;
+        std::vector<InputShape> inputShapes;
+        std::vector<ov::Shape> targetShapes;
+        std::tie(inType, inputShapes, targetShapes) = obj.param;
+        std::ostringstream result;
+        result << "IS=";
+        for (const auto& shape : inputShapes) {
+            result << ov::test::utils::partialShape2str({shape.first}) << "_";
+        }
+        result << "TS=";
+        for (const auto& shape : inputShapes) {
+            result << "(";
+            if (!shape.second.empty()) {
+                for (const auto& itr : shape.second) {
+                    result << ov::test::utils::vec2str(itr);
+                }
+            }
+            result << ")";
+        }
+        result << "Prc=" << inType;
+        return result.str();
+    }
+    void SetUp() override {
+        targetDevice = ov::test::utils::DEVICE_CPU;
+        configuration.insert({"ENFORCE_BF16", "YES"});
+        ElementType inType;
+        std::vector<InputShape> inputShapes;
+        std::vector<ov::Shape> targetShapes;
+        std::tie(inType, inputShapes, targetShapes) = this->GetParam();
+
+        if (inType == ElementType::bf16) {
+            configuration.insert({"ENFORCE_BF16", "YES"});
+        }
+        init_input_shapes(inputShapes);
+        auto input_params = ngraph::builder::makeDynamicParams(ov::element::f32, inputDynamicShapes);
+        auto zero = ov::op::v0::Constant::create(ov::element::i32, ov::Shape{}, {0});
+        auto one = ov::op::v0::Constant::create(ov::element::i32, ov::Shape{}, {1});
+        auto shapeof = std::make_shared<ov::op::v3::ShapeOf>(input_params[0], ov::element::i32);
+        auto gather = std::make_shared<ov::op::v8::Gather>(shapeof, one, zero);
+        auto convert = std::make_shared<ov::op::v0::Convert>(gather, ov::element::f32);
+        auto start = std::make_shared<ov::op::v0::Constant>(ov::element::f32, ov::Shape{}, 0);
+        auto step = std::make_shared<ov::op::v0::Constant>(ov::element::f32, ov::Shape{}, 1);
+        auto range = std::make_shared<ov::op::v4::Range>(start, convert, step, ov::element::f32);
+        auto matmul = std::make_shared<ov::op::v0::MatMul>(range, input_params[1], false, true);
+        auto result = std::make_shared<ov::op::v0::Result>(matmul);
+        result->set_friendly_name("output");
+        ov::ResultVector output_results = {result};
+
+        function = std::make_shared<ov::Model>(output_results, input_params, "remove_bf16_reorder");
+    };
+};
+
+namespace {
+TEST_P(RemoveUselessReorderCPUTest, CompareWithRefs) {
+    run();
+    CheckNumberOfNodesWithType(compiledModel, "Reorder", 4);
```

Comment:
Probably it is better to check the output values. The reference could be from template plugin, or from fp32 inference, or some precalculated numbers.
Of course, we need override generate_inputs method to feed inputs with dedicated numbers.

Reply:
yes, run will check the result value.